pkg/sdkserver/sdkserver.go

```diff
@@ -762,20 +792,190 @@ func (s *SDKServer) GetCounter(ctx context.Context, in *alpha.GetCounterRequest)
 	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
 		return nil, errors.Errorf("%s not enabled", runtime.FeatureCountsAndLists)
 	}
-	// TODO(#2716): Implement me
-	return nil, errors.Errorf("Unimplemented -- GetCounter coming soon")
+
+	s.logger.WithField("name", in.Name).Info("Getting Counter")
+
+	gs, err := s.gameServer()
```

Comment:
From conversations we had today, this is the logic I would propose for `GetCounter()`:

1. Grab `s.gameServer()` and all it's Count information.
2. Check a local copy of what was last successfully updated to K8s for count intformation. If it's `ObjectMeta.Generation` value is greater than that of `s.gameServer()` use that information instead.
3. If there is any batch data waiting, then apply that batch data to what you got from (1) and (2).
4. We now should have the most up to date count information, and capacity information we could possibly get.
5. If the count value is less than 0, make it zero
6. If the count value is greater than capacity, make it the capacity.
7. Return the result.

Finally, this logic (maybe in a separate private function without locking, so each public function can apply it's own locking) can then be reused in `UpdateCounter()` to determine if we'll be letting the request the user has sent into the batch queue or not.

Reply:
For step 2 I added this in the s.gameServer() logic. Are you OK with putting it there, or prefer not to change that method?

Reply:
That's a _really_ good question. I'm fairly sure that that is actually going to be fine - since we have the `generation` check.

it would also mean that if we want any other SDK feature to be update-to-date immediately on `s.gameServerGetter.GameServers.(..).Update(ctx, gsCopy, ...)`, it's super easy to do (update the local version we have), and those that we don't care about (which is most other functions), if you don't update the local version, it's fine.

So let's run with it, I think it will work, and the abstraction could be useful down the line.